src/main/java/com/cse/helium/app/controllers/VersionController.java

```diff
@@ -22,24 +24,34 @@
   @Autowired
   ApplicationContext context;
 
+  @Autowired
+  SwaggerConfig swaggerConfig;
+
   /**
    * Returns the application build version.
    *
    * @param response ServerHttpResponse passed into the alternate version handler by Spring
-   * @return Mono/<String/> container the build number
+   * @return Mono{@literal <}Map{@literal <}String, 
+   *      String{@literal <}{@literal <} container the build number
   */
   @GetMapping(name = "Helium Version Controller",
       value = "/version",
-      produces = MediaType.TEXT_PLAIN_VALUE)
-  public Mono<String> version(ServerHttpResponse response) {
+      produces = MediaType.APPLICATION_JSON_VALUE)
+  public Mono<Map<String, String>> version(ServerHttpResponse response) {
+    String apiVersion = "1.0";
     try {
-      response.setStatusCode(HttpStatus.OK);
-      return Mono.just(context.getBean(BuildConfig.class).getBuildVersion());
+      apiVersion = swaggerConfig.getInfo().get("version");      
     } catch (Exception ex) {
-
-      logger.error("Error recieved in VersionController", ex);
-      return Mono.error(new ResponseStatusException(
-        HttpStatus.INTERNAL_SERVER_ERROR, "version Error"));
+      logger.error("Error received in VersionController.", ex);
     }
+
+    // build the json result body
+    LinkedHashMap<String, String> versionResult = new LinkedHashMap<>();
+
+    versionResult.put("appVersion", context.getBean(BuildConfig.class).getBuildVersion());
+    versionResult.put("apiVersion", apiVersion);
+
+    response.setStatusCode(HttpStatus.OK);
+    return Mono.just(versionResult);
   }
```

Comment:
Should we continue to return the error if getInfo() throws and error?  Should we remove the try-catch or expand it to include other lines the method?

Let's chat for 5 min on it.